ndd_slurm.py

```diff
@@ -184,42 +190,87 @@ def get_slaves(args):
         raise ValueError('one of -D or -d must be specified')
 
 
-def run_slave(args):
-    procs = []
+def run_slave_tar(args, procs):
+    read_fd, write_fd = os.pipe()
+    procs.append(init_process(['tar', '-xC', args.o], read_fd=read_fd))
+    return write_fd
+
+
+def run_slave_comp(args, procs):
```

Comment:
`comp` may be confusing, let it be something like `start_slave_decompressor` or even `start_slave_unpack` because we not only are decompressing but may start untar here.